File: app/src/androidTest/java/org/bspb/smartbirds/pro/androidTest/internal/BatteryNotification.java
package org.bspb.smartbirds.pro.androidTest.internal;

import androidx.test.core.app.ApplicationProvider;

import org.bspb.smartbirds.pro.prefs.SmartBirdsPrefs_;
import org.bspb.smartbirds.pro.prefs.UserPrefs_;
import org.junit.runners.model.Statement;

/**
 * Test helper to set logged in or logged out state - it is tightly coupled with how the app
 * handles the authentication so the tests are not
 */
public class BatteryNotification extends Statement {
    private final boolean shown;

    public BatteryNotification(boolean shown) {
        this.shown = shown;
    }

    @Override
    public void evaluate() throws Throwable {
        new SmartBirdsPrefs_(ApplicationProvider.getApplicationContext())
                .edit()
                .isBatteryOptimizationDialogShown().put(shown)
                .apply();

    }
}

File: app/src/androidTest/java/org/bspb/smartbirds/pro/tools/rule/internal/BatteryNotification.kt
package org.bspb.smartbirds.pro.tools.rule.internal

import androidx.test.core.app.ApplicationProvider
import org.bspb.smartbirds.pro.prefs.SmartBirdsPrefs
import org.junit.runners.model.Statement

class BatteryNotification(private val shown: Boolean) : Statement() {

    override fun evaluate() {
        with(SmartBirdsPrefs(ApplicationProvider.getApplicationContext())) {
            setBatteryOptimizationDialogShown(shown)
        }
    }

}